tests/evd/test_evd.py

```diff
@@ -0,0 +1,441 @@
+    length = t.shape[0]
+    C = np.ones((length, length), dtype=np.complex64)
+    for ii in range(length):
+        for jj in range(ii + 1, length):
+            gamma = (gamma0 - gamma_inf) * np.exp((t[ii] - t[jj]) / Tau0) + gamma_inf
+            C[ii, jj] = gamma * np.exp(1j * (ph[ii] - ph[jj]))
+            C[jj, ii] = np.conj(C[ii, jj])
+
+    return C
+
+
+def simulate_phase_timeSeries(
+    time_series_length=365, acquisition_interval=12, signal_rate=1, std_random=0, k=1
+):
+    # time_series_length: length of time-series in days
+    # acquisition_interval: time-differense between subsequent acquisitions (days)
+    # signal_rate: linear rate of the signal (rad/year)
+    # k: seasonal parameter, 1 for annaual and  2 for semi-annual
+    t = np.arange(0, time_series_length, acquisition_interval)
+    signal_phase = signal_rate * (t - t[0]) / 365.0
+    if k > 0:
+        seasonal = np.sin(2 * np.pi * k * t / 365.0) + np.cos(2 * np.pi * k * t / 365.0)
+        signal_phase = signal_phase + seasonal
+
+    # adding random temporal signal (which simulates atmosphere + DEM error + ...)
+    signal_phase = signal_phase + std_random * np.random.randn(len(t))
+    signal_phase = signal_phase - signal_phase[0]
+    # wrap the phase to -pi to p
+    signal_phase = np.angle(np.exp(1j * signal_phase))
+
+    return signal_phase, t
+
+
+def covarinace(C1, C2):
+
+    A1 = np.sum(np.abs(C1) ** 2)
+    A2 = np.sum(np.abs(C2) ** 2)
+
+    cov = np.sum(C1 * np.conjugate(C2)) / (np.sqrt(A1) * np.sqrt(A2))
+    return cov
+
+
+def compute_covariance_matrix(neighbor_stack):
+    numberOfSlc = neighbor_stack.shape[0]
+    cov_mat = np.zeros((numberOfSlc, numberOfSlc), dtype=np.complex64)
+    for ti in range(numberOfSlc):
+        for tj in range(ti + 1, numberOfSlc):
+            cov = covarinace(neighbor_stack[ti, :], neighbor_stack[tj, :])
+            cov_mat[ti, tj] = cov
+            cov_mat[tj, ti] = np.conjugate(cov)
+        cov_mat[ti, ti] = 1.0
+
+    return cov_mat
+
+
+def estimate_evd(cov_mat):
+
+    # estimate the wrapped phase based on the eigen value decomposition of the covariance matrix
+    w, v = np.linalg.eigh(cov_mat)
+
+    # the last eignevalue is the maximum eigenvalue
+    # However let's check to make sure
+    ind_max = np.argmax(w)
+
+    # the eignevector corresponding to the largest eigenvalue
+    # of the covariance matrix is the solution
+    evd_estimate = v[:, ind_max]
+
+    # refernce to the first acquisition
+    evd_estimate = evd_estimate * np.conjugate(evd_estimate[0])
+
+    return evd_estimate
+
+
+def simulate_bit_mask(Ny, Nx, filename="neighborhood_map"):
+
+    # number of uint32 bytes needed to store weights
+    number_of_bytes = np.ceil((Ny * Nx) / 32)
+
+    flags = np.ones((Ny, Nx), dtype=np.bool_)
+    flag_bits = np.zeros((Ny, Nx), dtype=np.uint8)
+
+    for ii in range(Ny):
+        for jj in range(Nx):
+            flag_bits[ii, jj] = flags[ii, jj].astype(np.uint8)
```

Comment:
```suggestion
    flag_bits = np.array(flags, dtype=np.uint8)
```
Can skip conversion in explicit loop.

`flags` and `flag_bits` do not appear to be used.